demo_nodes_cpp/src/topics/talker_raw.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "rcutils/cmdline_parser.h"
+#include "rcutils/snprintf.h"
+
+#include "std_msgs/msg/string.hpp"
+
+#include "rmw/raw_message.h"
+
+using namespace std::chrono_literals;
+
+void print_usage()
+{
+  printf("Usage for talker app:\n");
+  printf("talker [-t topic_name] [-h]\n");
+  printf("options:\n");
+  printf("-h : Print this help function.\n");
+  printf("-t topic_name : Specify the topic on which to publish. Defaults to chatter.\n");
+}
+
+class RawTalker : public rclcpp::Node
+{
+public:
+  explicit RawTalker(const std::string & topic_name)
+  : Node("raw_talker")
+  {
+    raw_msg_ = rmw_get_zero_initialized_raw_message();
+    auto allocator = rcutils_get_default_allocator();
+    rmw_initialize_raw_message(
+      &raw_msg_,
+      0u,
+      &allocator);
+
+    // Create a function for when messages are to be sent.
+    auto publish_message =
+      [this]() -> void
+      {
+        auto string_msg = std::make_shared<std_msgs::msg::String>();
+        string_msg->data = "Hello World:" + std::to_string(count_++);
+        auto string_ts =
+          rosidl_typesupport_cpp::get_message_type_support_handle<std_msgs::msg::String>();
+        rmw_raw_message_resize(&raw_msg_, 8u + static_cast<unsigned int>(string_msg->data.size()));
```

Comment:
What's the extra 8 bytes for?

Reply:
Is calling `rmw_raw_message_resize()` here necessary? It looks like `rmw_fastrtps`'s implementation  of `rmw_serialize` will resize if necessary.

https://github.com/ros2/rmw_fastrtps/pull/186/files#diff-b260f922d969aa4536762ca6b4dfe86dR51

Reply:
yeah. Maybe I should make the demo a bit specific. The idea here is that with preallocating the memory, no dynamic memory allocation is needed anymore and the complete pipeline down to the actual publishing is free of any memory allocation.
I'll change the code to make that more visible.